gpiozero/pins/pigpiod.py

```diff
@@ -99,133 +157,96 @@ class PiGPIOPin(Pin):
+        self._pwm = False
+        self._bounce = None
+        self._when_changed = None
+        self._callback = None
+        self._edges = pigpio.EITHER_EDGE
         try:
-            return cls._PINS[(host, port, number)]
-        except KeyError:
-            self = super(PiGPIOPin, cls).__new__(cls)
-            cls.pi_info(host, port) # implicitly creates connection
-            self._connection, self._pi_info = cls._CONNECTIONS[(host, port)]
-            try:
-                self._pi_info.physical_pin('GPIO%d' % number)
-            except PinNoPins:
-                warnings.warn(
-                    PinNonPhysical(
-                        'no physical pins exist for GPIO%d' % number))
-            self._host = host
-            self._port = port
-            self._number = number
-            self._pull = 'up' if self._pi_info.pulled_up('GPIO%d' % number) else 'floating'
-            self._pwm = False
-            self._bounce = None
-            self._when_changed = None
-            self._callback = None
-            self._edges = pigpio.EITHER_EDGE
-            try:
-                self._connection.set_mode(self._number, pigpio.INPUT)
-            except pigpio.error as e:
-                raise ValueError(e)
-            self._connection.set_pull_up_down(self._number, self.GPIO_PULL_UPS[self._pull])
-            self._connection.set_glitch_filter(self._number, 0)
-            cls._PINS[(host, port, number)] = self
-            return self
-
-    def __repr__(self):
-        if self._host == 'localhost':
-            return "GPIO%d" % self._number
-        else:
-            return "GPIO%d on %s:%d" % (self._number, self._host, self._port)
-
-    @property
-    def host(self):
-        return self._host
-
-    @property
-    def port(self):
-        return self._port
-
-    @property
-    def number(self):
-        return self._number
+            self.factory.connection.set_mode(self.number, pigpio.INPUT)
+        except pigpio.error as e:
+            raise ValueError(e)
+        self.factory.connection.set_pull_up_down(self.number, self.GPIO_PULL_UPS[self._pull])
+        self.factory.connection.set_glitch_filter(self.number, 0)
 
     def close(self):
-        # If we're shutting down, the connection may have disconnected itself
-        # already. Unfortunately, the connection's "connected" property is
-        # rather buggy - disconnecting doesn't set it to False! So we're
-        # naughty and check an internal variable instead...
-        if self._connection.sl.s is not None:
+        if self.factory.connection:
             self.frequency = None
             self.when_changed = None
             self.function = 'input'
-            self.pull = 'up' if self._pi_info.pulled_up('GPIO%d' % self.number) else 'floating'
+            self.pull = 'up' if self.factory.pi_info.pulled_up('GPIO%d' % self.number) else 'floating'
+
+    def _get_address(self):
```

Comment:
This appears to be identical to the parent-class (PiPin) version, so could be removed from here?